codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -214,57 +214,33 @@ let getMostBehindPartitions = (
   ->Js.Array.slice(~start=0, ~end_=maxNumQueries)
 }
 
-let updatePartition = (self: t, ~fetchState: FetchState.t, ~partitionId: partitionIndex) => {
-  {...self, partitions: self.partitions->Utils.Array.setIndexImmutable(partitionId, fetchState)}
-}
-
 type nextQueries = WaitForNewBlock | NextQuery(array<FetchState.nextQuery>)
+
 /**
 Gets the next query from the fetchState with the lowest latestFetchedBlock number.
 */
-let getNextQueriesOrThrow = (
-  self: t,
-  ~currentBlockHeight,
-  ~maxPerChainQueueSize,
-  ~partitionsCurrentlyFetching,
-) => {
-  let optUpdatedPartition = ref(None)
-  let includesWaitForNewBlock = ref(false)
+let getNextQueries = (self: t, ~maxPerChainQueueSize, ~partitionsCurrentlyFetching) => {
   let nextQueries = []
+  let updatedPartitions = Js.Dict.empty()
```

Comment:
I don't really see the purpose of returning this dict? Why not return the fully updated PartitionedFetchState?

Reply:
No particular reason, but we can skip the state update when there are no merges